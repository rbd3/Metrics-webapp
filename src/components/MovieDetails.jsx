import React from 'react';
import { Link, useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import '../styles/MovieDetails.css';

function MovieDetails() {
  const { id } = useParams();
  const movies = useSelector((state) => state.movies.Movies);

  const movie = movies.find((movie) => movie.id.toString() === id);

  if (!movie) {
    return <h2>Loading...</h2>;
  }

  const {
    name, rating, summary, image, language, category, seasons,
  } = movie;
  const Summary = summary
    .replace(/<p>/g, '')
    .replace(/<\/p>/g, '')
    .replace(/<b>/g, '')
    .replace(/<\/b>/g, '');

  return (
    <div className="movie-details">
      <Link className="back-button" to="/">
        <i className="fa-solid fa-arrow-left" />
        Back
      </Link>
      <div className="details-container">
        <div className="headings">
          <h2 className="name-details">{name}</h2>
          <div className="rating">
            <h3>
              Rating:
              {rating.average}
            </h3>
          </div>
        </div>
        <div className="movie-main">
          <img src={image.medium} alt="" />
          <div className="summary-section">
            <p className="summary">{Summary}</p>
            <p className="season">
              Seasons:
              {' '}
              {seasons}
            </p>
            <p className="language">
              Language:
              {' '}
              {language}
            </p>
            <p className="type">
              Category:
              {' '}
              {category.join(', ')}
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}

export default MovieDetails;
