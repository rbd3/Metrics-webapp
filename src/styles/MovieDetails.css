.movie-details {
  padding: 20px 10px;
  display: flex;
  flex-direction: column;
  background-color: rgb(243, 204, 211);
}

.back-button {
  text-decoration: none;
  padding: 1rem 2rem;
  color: black;
  font-size: 2rem;
  font-weight: bold;
  border-radius: 1rem;
  box-shadow: 1px 5px 5px #533333;
  background-color: rgb(247, 157, 172);
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
  max-width: 10%;
  align-items: center;
}

.details-container {
  display: flex;
  flex-direction: column;
  padding: 25px;
  margin: 5px auto;
  background-color: #f194b8;
  box-shadow: 1px 5px 5px #533333;
  color: rgb(20, 19, 19);
}

.headings {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.movie-main {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.summary-section {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-left: 3%;
}

.summary {
  color: rgb(14, 11, 11);
  font-size: 1.3rem;
  font-weight: bold;
  margin-bottom: 2%;
}

.movie-categories {
  display: flex;
  flex-direction: row;
}

.movie-categories span {
  margin-right: 20px;
}

.season {
  color: rgb(14, 11, 11);
  font-size: 1.4rem;
  font-weight: bold;
  margin-bottom: 1%;
}

.type {
  font-size: 1.4rem;
  font-weight: bold;
  text-decoration: none;
  color: rgb(14, 11, 11);
}

@media screen and (max-width: 768px) {
  .movie-main {
    flex-direction: column;
  }

  .movie-categories {
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .back-button {
    max-width: 30%;
  }
}
