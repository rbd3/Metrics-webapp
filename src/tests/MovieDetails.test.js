import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import MovieDetails from '../components/MovieDetails';

// Create a mock Redux store
const mockStore = configureStore([]);
const store = mockStore({
  movies: {
    Movies: [
      {
        id: '1',
        name: 'Movie Name',
        rating: {
          average: 8.5,
        },
        category: ['Drama', 'Thriller'],
        summary: 'Summary of the show',
        image: {
          medium: 'abc.jpg',
        },
        official: 'https://official-website.com',
        language: 'English',
      },
    ],
  },
});

test('renders MovieDetail component', () => {
  const { container } = render(
    <Provider store={store}>
      <Router>
        <MovieDetails />
      </Router>
    </Provider>,
  );

  expect(container.innerHTML).toMatchSnapshot();
});
