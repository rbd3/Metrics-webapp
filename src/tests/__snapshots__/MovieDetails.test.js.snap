// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieDetail component 1`] = `"<h2>Loading...</h2>"`;
